src/modules/git_state.rs

```diff
@@ -0,0 +1,165 @@
+use ansi_term::Color;
+use git2::{Repository, RepositoryState};
+use std::path::Path;
+
+use super::{Context, Module};
+
+/// Creates a module with the state of the git repository at the current directory
+///
+/// During a git operation it will show: REBASING, BISECTING, MERGING, etc.
+/// If the progress information is available (e.g. rebasing 3/10), it will show that too.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let repo_root = context.repo_root.as_ref()?;
+    let mut repository = Repository::open(repo_root).ok()?;
+    let state_description = get_state_description(&mut repository);
+
+    if let StateDescription::Clean = state_description {
+        return None;
+    }
+
+    let mut module = context.new_module("git_state")?;
+    module.get_prefix().set_value("(");
+    module.get_suffix().set_value(") ");
+    module.set_style(Color::Yellow.bold());
+
+    let label = match state_description {
+        StateDescription::Label(label) => label,
+        StateDescription::LabelAndProgress(label, _) => label,
+        // Should only be possible if you've added a new variant to StateDescription
+        _ => panic!("Expected to have a label at this point in the control flow."),
+    };
+
+    module.new_segment(label.segment_name, label.message_default);
+
+    if let StateDescription::LabelAndProgress(_, progress) = state_description {
+        module.new_segment("progress_current", &format!(" {}", progress.current));
+        module.new_segment("progress_divider", "/");
+        module.new_segment("progress_total", &format!("{}", progress.total));
```

Comment:
IIRC, if there's a TOML option by the name, then the value is overridden with that value. Does that mean we've created (unintended, undocumented) TOML options here which could break the module if they were set?

Reply:
It's not a big deal if the answer is "yes," it's just going to be a mental note in the 1.0 refactor to have ways to avoid this.

Reply:
Yeah, the answer is yes. I think this issue affects most modules in some form or another.

Reply:
Alright, we'll kick that particular can down the road for now then.

Reply:
Haha. You have uncovered my janky implementation of configuration. 🙃 
I would definitely like to rework it during the refactor.